src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -99,6 +102,22 @@ public class NodeSettings
                 this.Network = testNet ? Network.TestNet : regTest ? Network.RegTest : Network.Main;
             }
 
+            // Setting the data directory.
```

Comment:
what's the advantage of moving this here?

Reply:
The variables on which that code depends have been made read-only. This makes it possible to execute that code earlier - in the constructor. The fact that the DataFolder is then set earlier also makes it possible to postpone LoadConfiguration which would otherwise had to be called earlier (as previously) just to ensure that the DataFolder is set in time for the code that depends on it.